src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -0,0 +1,235 @@
+  private long lineNumber = 0;
+
+  private final boolean isLastWorker;
+  private final long fileSize;
+
+  private long byteOverlap = 10;
+  private long startByteRange;
+  private long endByteRange;
+  long workerID;
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * The logger for debug, trace, etc. messages in this class.
+   */
+  private static final org.slf4j.Logger LOGGER = org.slf4j.LoggerFactory.getLogger(CSVFileScanFragment.class);
+
+  public CSVFileScanFragment(final String filename, final Schema schema, final int workerID, final int totalWorkers) {
+    this(filename, schema, workerID, totalWorkers, null, null, null, null);
+  }
+
+  public CSVFileScanFragment(final DataSource source, final Schema schema, final int workerID, final int totalWorkers) {
+    this(source, schema, workerID, totalWorkers, null, null, null, null);
+  }
+
+  public CSVFileScanFragment(final String filename, final Schema schema, final int workerID, final int totalWorkers,
+      final Character delimiter) {
+    this(new FileSource(filename), schema, workerID, totalWorkers, delimiter, null, null, null);
+  }
+
+  public CSVFileScanFragment(final DataSource source, final Schema schema, final int workerID, final int totalWorkers,
+      final Character delimiter) {
+    this(source, schema, workerID, totalWorkers, delimiter, null, null, null);
+  }
+
+  public CSVFileScanFragment(final String filename, final Schema schema, final int workerID, final int totalWorkers,
+      @Nullable final Character delimiter, @Nullable final Character quote, @Nullable final Character escape,
+      @Nullable final Integer numberOfSkippedLines) {
+    this(new FileSource(filename), schema, workerID, totalWorkers, delimiter, quote, escape, numberOfSkippedLines);
+  }
+
+  public CSVFileScanFragment(final DataSource source, final Schema schema, final int workerID, final int totalWorkers,
+      @Nullable final Character delimiter, @Nullable final Character quote, @Nullable final Character escape,
+      @Nullable final Integer numberOfSkippedLines) {
+    this.source = (AmazonS3Source) Preconditions.checkNotNull(source, "source");
+    this.schema = Preconditions.checkNotNull(schema, "schema");
+
+    this.delimiter = MoreObjects.firstNonNull(delimiter, CSVFormat.DEFAULT.getDelimiter());
+    this.quote = MoreObjects.firstNonNull(quote, CSVFormat.DEFAULT.getQuoteCharacter());
+    this.escape = escape != null ? escape : CSVFormat.DEFAULT.getEscapeCharacter();
+    this.numberOfSkippedLines = MoreObjects.firstNonNull(numberOfSkippedLines, 0);
+    this.workerID = workerID;
+
+    isLastWorker = workerID == totalWorkers;
+    fileSize = ((AmazonS3Source) source).getFileSize();
+
+    if (fileSize < 10 && workerID == 1) {
+      startByteRange = 0;
+      endByteRange = fileSize;
+    } else {
+      long partitionSize = fileSize / totalWorkers;
+      startByteRange = (partitionSize) * (workerID - 1);
```

Comment:
If workers are not consecutive this would mean that the coordinator should be assigning the workers with a ```workerID```. I really tried to keep the bulk of the logic worker independent, but now it seems easier for the coordinator to do this work. 

Another related problem I had while testing this was when I tried to ingest into a specific range of workers (i.e. worker IDs in ranges 11-20). The current implementation breaks if we try to do this. 

Again, because of this, I'm thinking about moving this logic of assigning workerIDs to the coordinator.